src/frontend/src/optimizer/plan_node/logical_project.rs

```diff
@@ -148,6 +148,14 @@ impl fmt::Display for LogicalProject {
 
 impl ColPrunable for LogicalProject {
     fn prune_col(&self, required_cols: &[usize], ctx: &mut ColumnPruningContext) -> PlanRef {
+        // for example, in query:
+        // `insert into t values(v1, v2) returning v1;`
+        // column pruning may mess up the insert values.
+        // so column pruning is disabled in insert->project
+        if self.input().node_type() == PlanNodeType::LogicalInsert {
+            return self.clone().into();
+        }
```

Comment:
I think we can not do it, especially in the `Project::prune_col`. Instead, we need to change the `Insert::prune_col`. 
e.g. for the `insert into t values(v1, v2) returning v2*2;` we will have the following plan 
```SQL
Project(input_ref(1)*2)
    Insert(v1, v2)
         Values(v1, v2)
```
and the `Project::prune_col` will call its child, `Insert::prune_col` with the required column [1]. the Insert::prune_col need to return a plan only with the column 0. the plan is expected to be 
```SQL
Project(input_ref(0)*2)
    Project(input_ref(1))
        Insert(v1, v2)
            Values(v1, v2)
```

You can refer to https://github.com/risingwavelabs/risingwave/blob/main/src/frontend/src/optimizer/plan_node/logical_project_set.rs#L221-L230

Reply:
indeed, modified.